### PR TITLE
Integ specs views

### PR DESCRIPTION
**Pull request by OmarMWarraich**
# Worthy Reviewer,

## Kindly approve the following project requirements.

- Improve Performance Through the Use of Eager Loading.
- Integration Specs with Capybara.

_**All Tests are Passing**_
_**Screen Shot For Spec Results**_
Screenshot from 2023-02-03 17-26-36

### Kindly note that the unused helper modules have been excluded from the tests.

 _**Screen Shots for Query Reduction**_
Screenshot from 2023-02-01 21-33-35
Screenshot from 2023-02-01 13-25-00
Screenshot from 2023-02-01 21-33-59
Screenshot from 2023-02-01 13-25-32
Screenshot from 2023-02-01 21-38-03
Screenshot from 2023-02-01 15-40-04
Screenshot from 2023-02-01 16-47-41

_**UI**_
Screenshot from 2023-02-03 15-03-44
Screenshot from 2023-02-03 15-03-51
Screenshot from 2023-02-03 15-03-59
Screenshot from 2023-02-03 15-04-20
Screenshot from 2023-02-03 15-04-42
Screenshot from 2023-02-03 15-04-54
Screenshot from 2023-02-03 15-05-18

### CODE REVIEW

File: spec/features/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    expect(page).to have_content(@posts[0].comments_counter)
+  end
+
+  it 'shows the number of likes for the post' do
+    expect(page).to have_content(@posts[0].likes_counter)
+  end
+
+  it 'shows the pagination section' do
+    expect(page).to have_content('Pagination')
+  end
```
Comment (Nemwel-Boniface) [inline spec/features/posts_index_spec.rb:49]: - [x] You seem to have forgotten to add the tests for "I can see a post's title". Kindly include it.

image

- [x] You seem to have forgotten to add the tests for "I can see the first comments on a post"

image

- [x] You seem to have forgotten to add the tests for "When I click on a post, it redirects me to that post's show page."

image

File: spec/features/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  it "should show the post's comments counter" do
+    expect(page).to have_content(@post1.comments_counter)
+  end
+
+  it "should show the post's likes counter" do
+    expect(page).to have_content(@post1.likes_counter)
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/features/posts_show_spec.rb]: - [x] You seem to have forgotten to add the tests for "I can see the username of each commenter". Kindly include it.

image

- [x] You seem to have forgotten to add the tests for "I can see the comment each commentator left". KIndly include it.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @OmarMWarraich        :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/posts_index_spec.rb (hunk 1 above)
Comment (OmarMWarraich) [reply spec/features/posts_index_spec.rb:49]: @Nemwel-Boniface My bad. I have completed these requirements.

File: spec/features/posts_show_spec.rb (hunk 2 above)
Comment (OmarMWarraich) [reply spec/features/posts_show_spec.rb]: @Nemwel-Boniface Completed these as well.

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @OmarMWarraich  , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: aaa
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R
- [x] Tests are passing 








 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**